esnet-security/SCRAM
# tests: add coverage and fix ci

This MR adds in tests for the past ~year of changes. It also fixes a bunch of issues that were preventing CI from running properly sometimes. 

## Test Related Work
* Add new tests to make sure that our ASN validation is properly covered.
  * Make sure that the appropriate warning is being raised (i.e. not just failing randomly)
* Fixed tests that were silently failing for the `We can block an IP` Scenario
* Added better logging to ASN validation
* Added pattern for raising custom exceptions and setting aside code that might some day be shared between various translators files. 
## CI Fixes
* Now we remove orphaned containers from when you run tests (start test runs with --rm flag)
* Gitlab jobs now clean up properly. Before, since we were running cleanup as a separate stage, stale containers would be left behind if that stage was picked up by a different runner. Then when you run a CI job again on that runner, the TCP ports specified in compose are still in use so the job fails.
